src/services/utilities.ts

```diff
@@ -1992,6 +1992,10 @@ namespace ts {
         return typeIsAccessible ? res : undefined;
     }
 
+    export function isDecimalIntegerLiteral(node: Node): node is NumericLiteral {
+        return isNumericLiteral(node) && /^\d+$/.test(node.text);

```

Comment:
this actually doesn't do what it's supposed to do. `NumericLiteral#text` contains the normalized value, e.g. `1.0` -> `1`

you probably want to use `.getText()` here to get the raw source text of the numeric literal